integration_test/dapp_tests/steak/SteakTests.js

```diff
@@ -0,0 +1,175 @@
+      epoch_period: 259200,
+      unbond_period: 1814400,
+      validators: validators.slice(0, 3),
+    };
+    const contractAddresses = await instantiateHubContract(
+      hubCodeId,
+      ownerAddress,
+      instantiateMsg,
+      "steakhub"
+    );
+
+    // Deploy pointer for token contract
+    const pointerAddr = await deployErc20PointerForCw20(
+      hre.ethers.provider,
+      contractAddresses.tokenContract
+    );
+    const tokenPointer = new hre.ethers.Contract(
+      pointerAddr,
+      ABI.ERC20,
+      hre.ethers.provider
+    );
+
+    return {
+      hubAddress: contractAddresses.hubContract,
+      tokenAddress: contractAddresses.tokenContract,
+      tokenPointer,
+    };
+  }
+
+  async function testBonding(address, amount) {
+    const initialBalance = await queryTokenBalance(tokenAddress, address);
+    expect(initialBalance).to.equal("0");
+
+    await bond(hubAddress, address, amount);
+    const tokenBalance = await queryTokenBalance(tokenAddress, address);
+    expect(tokenBalance).to.equal(`${amount}`);
+  }
+
+  async function testUnbonding(address, amount) {
+    const initialBalance = await queryTokenBalance(tokenAddress, address);
+    const response = await unbond(hubAddress, tokenAddress, address, amount);
+    expect(response.code).to.equal(0);
+
+    // Balance should be updated
+    const tokenBalance = await queryTokenBalance(tokenAddress, address);
+    expect(tokenBalance).to.equal(`${Number(initialBalance) - amount}`);
+  }
+
+  before(async function () {
+    // Set up the owner account
+    owner = await setupAccount("steak-owner");
+
+    // Store and deploy contracts
+    ({ hubAddress, tokenAddress, tokenPointer } = await deployContracts(
+      owner.address
+    ));
+  });
+
+  describe("Bonding", async function () {
+    it("Associated account should be able to bond", async function () {
+      const amount = 1000000;
+      await testBonding(owner.address, amount);
+
+      // Verify that address is associated
+      const evmAddress = await getEvmAddress(owner.address);
+      expect(evmAddress).to.not.be.empty;
+
+      // Check pointer balance
+      const pointerBalance = await tokenPointer.balanceOf(evmAddress);
+      expect(pointerBalance).to.equal(`${amount}`);
+    });
+
+    it("Unassociated account should be able to bond", async function () {
+      const unassociatedAccount = await setupAccount("unassociated", false);
+      // Verify that account is not associated yet
+      const initialEvmAddress = await getEvmAddress(
+        unassociatedAccount.address
+      );
+      expect(initialEvmAddress).to.be.empty;
+
+      await testBonding(unassociatedAccount.address, 1000000);
+
+      // Account should now be associated
+      const evmAddress = await getEvmAddress(unassociatedAccount.address);
+      expect(evmAddress).to.not.be.empty;
+    });
+  });
+
+  describe("Unbonding", async function () {
+    it("Associated account should be able to unbond", async function () {
```

Comment:
Feels like this test can be merged in to the associated 'Bonding' test, since it also covers the bonding part.